File: 922.go
func sortArrayByParityII(nums []int) []int {
	j := 1
	for i := 0; i < len(nums); i += 2 {
		if nums[i]%2 != i%2 {
			for nums[i]%2 == nums[j]%2 {
				j += 2
			}
			nums[i], nums[j] = nums[j], nums[i]
		}
	}

	return nums
}

File: 922.rs
impl Solution {
    pub fn sort_array_by_parity_ii(mut nums: Vec<i32>) -> Vec<i32> {
        let mut i = 0;
        let mut j = 1;

        while i < nums.len() {
            if nums[i] % 2 != (i % 2) as i32 {
                while nums[j] % 2 == nums[i] % 2 {
                    j += 2;
                }

                nums.swap(i, j);
            }

            i += 2;
        }

        nums
    }
}
